Core/clim-basic/geometry/region-composition.lisp

```diff
@@ -239,25 +239,13 @@
 (defmethod region-intersection ((ellipse standard-ellipse) (line standard-line))
   (region-intersection line ellipse))
 
-(defmethod region-intersection ((a standard-line) (b standard-polygon))
+(defmethod region-intersection ((b polygon) (a line))
   (multiple-value-bind (x1 y1) (line-start-point* a)
     (multiple-value-bind (x2 y2) (line-end-point* a)
       (schnitt-line/polygon x1 y1 x2 y2 b))))
 
-(defmethod region-intersection ((b standard-polygon) (a standard-line))
-  (multiple-value-bind (x1 y1) (line-start-point* a)
-    (multiple-value-bind (x2 y2) (line-end-point* a)
-      (schnitt-line/polygon x1 y1 x2 y2 b))))
-
-(defmethod region-intersection ((a standard-line) (b standard-rectangle))
-  (multiple-value-bind (x1 y1) (line-start-point* a)
-    (multiple-value-bind (x2 y2) (line-end-point* a)
-      (schnitt-line/polygon x1 y1 x2 y2 b))))
-
-(defmethod region-intersection ((b standard-rectangle) (a standard-line))
-  (multiple-value-bind (x1 y1) (line-start-point* a)
-    (multiple-value-bind (x2 y2) (line-end-point* a)
-      (schnitt-line/polygon x1 y1 x2 y2 b))))
+(defmethod region-intersection ((a line) (b polygon))
+  (region-intersection b a))
```

Comment:
I like this cleanup, but the second method adds a generic function call. Would a local macro for generating the body (one as is and the second with `a` and `b` swapped) make sense?

Reply:
it is certainly possible, but I think that it will obfuscate code for little gain (i.e it is unlikely it will be a bottleneck in practice).

Reply:
We seem to have different tolerances for introducing performance regressions when cleaning up. My experience so far has been that we have to re-optimize sooner or later after introducing performance regressions to facilitate cleanups (e.g. `clx-fb`). I have also run into performance issues with the geometry module several times already, so I'm more worried than I would normally be.

In this case, reducing four methods to two still seems like a good simplification (with or without macrology).

Reply:
please look at the last commit